runtime/doc/lsp.txt

```diff
@@ -1647,6 +1659,27 @@ trigger()                                       *vim.lsp.completion.trigger()*
     Trigger LSP completion in the current buffer.
 
 
+==============================================================================
+Lua module: vim.lsp.folding_range                          *lsp-folding_range*
+
+foldclose({kind}, {winid})                 *vim.lsp.folding_range.foldclose()*
```

Comment:
The logic is maybe complex enough to live in its own module, but we probably don't need to expose it as such? Should this live at `vim.lsp.foldclose`, or do we plan to have many more fold-related functions in the future? E.g. `folding_range.enable()`, `folding_range.is_enabled`, etc.

Reply:
This is mainly about naming convention. I found that the functions under `vim.lsp.<name>`, apart from those directly related to LSP (such as `vim.lsp.buf_request`), only have functions related to specific methods when there are concrete interfaces in nvim. For example, `vim.lsp.tagfunc`, `vim.lsp.formatexpr`. The others are either in `vim.lsp.buf` or have their own module.

It is also possible to add additional functions. I hope to provide as few interfaces as possible initially to cover the functionality that LSP can offer. However, for convenience, it may also be possible to add a `foldopen()` in the future to correspond with `foldclose()`, or to add a function/hook to get `foldlevel` corresponding to vim's `set foldlevel`.

Reply:
If more functions are planned then I guess this is fine.